nutils/function.py

```diff
@@ -2452,6 +2452,27 @@ def partition(f: IntoArray, *levels: float) -> Sequence[Array]:
   steps = map(subtract, signs[:-1], signs[1:])
   return [.5 - .5 * signs[0]] + [.5 * step for step in steps] + [.5 + .5 * signs[-1]]
 
+def heaviside(f: IntoArray):
+  '''
+  Create a heaviside function based on a levelset:
+  `H(f) = { 0: f < 0; 0.5: f == 0; 1: f > 0`.
+
+  Heaviside is a special case of :func:`partition`.
+
+  Args
+  ----
+  f : :class:`Array`
+      Scalar-valued function
+
+  Returns
+  -------
+  :class:`Array`
+      The heaviside function.
+  '''
+
+  left, right = partition(f, 0)
```

Comment:
Since `partition` itself uses `sign`, wouldn't it make sense to use `.5 + .5 * sign(f)` instead?

Reply:
absolutely! :)